### PR TITLE
[FSSDK-8663] chore: General clean up before ticket work

### PR DESCRIPTION
**Pull request by mikechu-optimizely**
## Summary
- Rider and GitHub issues cleaned up

## Test plan
- All existing unit & end-to-end tests should pass. 

## Issues
- [FSSDK-8663](https://jira.sso.episerver.net/browse/FSSDK-8663)

### CODE REVIEW

File: OptimizelySDK.DemoApp/OptimizelySDK.DemoApp.csproj [hunk 1]
```diff
@@ -249,9 +248,6 @@
     <Content Include="Scripts\popper-utils.min.js.map" />
     <Content Include="Scripts\popper-utils.js.map" />
   </ItemGroup>
-  <ItemGroup>
-    <Folder Include="App_Data\" />
-  </ItemGroup>
```
Comment (mikechu-optimizely) [inline OptimizelySDK.DemoApp/OptimizelySDK.DemoApp.csproj:254]: Non-existent folder

File: OptimizelySDK.DemoApp/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -37,7 +37,7 @@
 					@Html.ActionLink("Home", "Index", "Home", new { area = "" }, new { @class = "nav-link" })
 				</li>
 				<li class="nav-item">
-					@Html.ActionLink("Demo", "Index", "Demo/Config", new { area = "" }, new { @class = "nav-link" })
```
Comment (mikechu-optimizely) [inline OptimizelySDK.DemoApp/Views/Shared/_Layout.cshtml:40]: Incorrect way to call the Config method in the DemoController

File: OptimizelySDK.DemoApp/packages.config [hunk 3]
```diff
@@ -15,7 +15,7 @@
   <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" targetFramework="net45" />
   <package id="Modernizr" version="2.6.2" targetFramework="net45" />
   <package id="murmurhash-signed" version="1.0.2" targetFramework="net45" />
-  <package id="Newtonsoft.Json" version="9.0.1" targetFramework="net45" />
```
Comment (mikechu-optimizely) [inline OptimizelySDK.DemoApp/packages.config:18]: Vulnerable version

File: OptimizelySDK.Net35/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -38,5 +38,5 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 [assembly: AssemblyVersion("4.0.0.0")]
-[assembly: AssemblyFileVersion("4.0.0.0-beta")]
```
Comment (mikechu-optimizely) [inline OptimizelySDK.Net35/Properties/AssemblyInfo.cs:41]: AssemblyFileVersion doesn't support meta tags in the version number

File: OptimizelySDK.Tests/App.config
Comment (mikechu-optimizely) [inline OptimizelySDK.Tests/App.config:1]: Automatic lint fixes

File: OptimizelySDK.sln [hunk 5]
```diff
@@ -16,10 +16,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 	ProjectSection(SolutionItems) = preProject
 		CHANGELOG.MD = CHANGELOG.MD
 		CONTRIBUTING.md = CONTRIBUTING.md
-		OptimizelySDK.Package\OptimizelySDK.nuspec = OptimizelySDK.Package\OptimizelySDK.nuspec
-		OptimizelySDK.Package\pack.ps1 = OptimizelySDK.Package\pack.ps1
 		README.md = README.md
-		OptimizelySDK.Package\verifysn.ps1 = OptimizelySDK.Package\verifysn.ps1
```
Comment (mikechu-optimizely) [inline OptimizelySDK.sln:22]: Moved template to repo root and consuming code in [Actions workflow](https://github.com/optimizely/csharp-sdk/blob/master/.github/workflows/csharp_release.yml#L168)

File: OptimizelySDK/Config/HttpProjectConfigManager.cs [hunk 6]
```diff
@@ -19,6 +19,7 @@
 #endif
 
 using System;
+using System.Linq;
```
Comment (mikechu-optimizely) [inline OptimizelySDK/Config/HttpProjectConfigManager.cs:22]: Need for lower Framework versions

**Review by mikechu-optimizely — COMMENTED** [review]
Developer code review & notes

File: OptimizelySDK.Tests/packages.config [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
-  <package id="ReportGenerator" version="2.5.6" targetFramework="net45"/>
-</packages>
+  <package id="Castle.Core" version="4.0.0" targetFramework="net45" />
+  <package id="Moq" version="4.7.1" targetFramework="net45" />
+  <package id="Newtonsoft.Json" version="13.0.2" targetFramework="net45" />
+  <package id="NUnit" version="2.6.4" targetFramework="net45" />
+  <package id="NUnitTestAdapter" version="2.1.1" targetFramework="net45" />
+  <package id="OpenCover" version="4.6.519" targetFramework="net45" />
+  <package id="ReportGenerator" version="2.5.6" targetFramework="net45" />
+</packages>
```
Comment (Mat001) [inline OptimizelySDK.Tests/packages.config:10]: extra line

File: OptimizelySDK/Config/HttpProjectConfigManager.cs [hunk 8]
```diff
@@ -201,12 +202,7 @@ protected override ProjectConfig Poll()
         {
             var datafile = GetRemoteDatafileResponse();
 
-            if (datafile == null)
-            {
-                return null;
-            }
-
-            return DatafileProjectConfig.Create(datafile, Logger, ErrorHandler);
+            return datafile == null ? null : DatafileProjectConfig.Create(datafile, Logger, ErrorHandler);
```
Comment (Mat001) [inline OptimizelySDK/Config/HttpProjectConfigManager.cs:205]: You're creating project config here, can you say why?

**Review by Mat001 — APPROVED** [review]
Looks ok (for my understanding of C#). Left a comment

File: OptimizelySDK/Config/HttpProjectConfigManager.cs [hunk 9]
```diff
@@ -201,12 +202,7 @@ protected override ProjectConfig Poll()
         {
             var datafile = GetRemoteDatafileResponse();
 
-            if (datafile == null)
-            {
-                return null;
-            }
-
-            return DatafileProjectConfig.Create(datafile, Logger, ErrorHandler);
```
Comment (mikechu-optimizely) [inline OptimizelySDK/Config/HttpProjectConfigManager.cs:209]: Thanks for the question @Mat001.  This was a refactor for clarity ... but perhaps it's not clear.  I'm moving it back.

**Pull request merged**